src/pybind/mgr/smb/handler.py

```diff
@@ -719,6 +719,27 @@ def _generate_share(
             'smbd profiling share': 'yes',
         }
     }
+
+    # Add QoS settings if they exist
+    if cephfs.qos:
+        qos = cephfs.qos
+        if qos.read_iops_limit:
+            cfg['options'][f'{ceph_vfs}:read_iops_limit'] = str(
```

Comment:
You can reduce the repetition a little by doing something like:
`opts = cfg['options']` at the same level as `qos = cephfs.qos`
if you want